package.json

```diff
@@ -181,7 +182,7 @@
     "json-stringify-safe": "5.0.1",
     "lodash": "^4.17.21",
     "lru-cache": "^4.1.5",
-    "minimatch": "^3.0.4",
```

Comment:
This is expression should equate to the same semantic version correct and that it's actually be resolved to `3.1.2` based on the yarn lock file? Can we verify other plugins don't define `^3.0.4` otherwise it would break the build. If they do we would could just leave this the same.

Reply:
@kavilla Does it mean changes go only into yarn.lock file and package.json will not be enforced with the upgraded version as it can break build on plugin dependencies unless we know there are no plugins depending on older version in the package.json file? I see this way of fixing other CVE dependencies as well.

Reply:
@manasvinibs if the problem is a direct dependency, of course updating the dep is necessary. When it is a nested dependency that needs bumping, sometimes you can get away with just changing the lockfile but sometimes you need to add overrides.

Here, bumping this dependency, doesn't do anything since in the lockfile, it is already resolved to `3.1.2`.

Reply:
I think we need to add a resolution `"**/minimatch": "^3.0.5"`. 

Therefore it could make a clean up and no impact to plugins.

```
[1/4] Why do we have the module "minimatch"...?
[2/4] Initialising dependency graph...
warning Resolution field "typescript@4.0.2" is incompatible with requested version "typescript@~4.5.2"
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "minimatch@3.1.2"
info Has been hoisted to "minimatch"
info Reasons this module exists
   - "workspace-aggregator-0e718540-3470-4772-97d3-335a2b03a99c" depends on it
   - Specified in "dependencies"
   - Hoisted from "_project_#mocha#minimatch"
   - Hoisted from "_project_#glob#minimatch"
   - Hoisted from "_project_#eslint#minimatch"
   - Hoisted from "_project_#eslint-plugin-import#minimatch"
   - Hoisted from "_project_#eslint-plugin-jsx-a11y#minimatch"
   - Hoisted from "_project_#eslint-plugin-node#minimatch"
   - Hoisted from "_project_#minimatch"
   - Hoisted from "_project_#eslint-plugin-react#minimatch"
   - Hoisted from "_project_#grunt#minimatch"
   - Hoisted from "_project_#tape#dotignore#minimatch"
   - Hoisted from "_project_#mocha#glob#minimatch"
   - Hoisted from "_project_#load-grunt-config#glob#minimatch"
   - Hoisted from "_project_#grunt#glob#minimatch"
   - Hoisted from "_project_#@osd#pm#multimatch#minimatch"
   - Hoisted from "_project_#archiver#readdir-glob#minimatch"
   - Hoisted from "_project_#nyc#test-exclude#minimatch"
   - Hoisted from "_project_#grunt#findup-sync#glob#minimatch"
   - Hoisted from "_project_#@percy#cli#@percy#cli-snapshot#serve-handler#minimatch"
   - Hoisted from "_project_#@osd#plugin-generator#ejs#jake#minimatch"
   - Hoisted from "_project_#@osd#plugin-generator#ejs#jake#filelist#minimatch"
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "156KB"
info Number of shared dependencies: 3
=> Found "@osd/ui-framework#minimatch@3.1.2"
info Reasons this module exists
   - "_project_#@osd#ui-framework#grunt-contrib-clean#rimraf#glob" depends on it
   - Specified in "dependencies"
   - Hoisted from "_project_#@osd#ui-framework#grunt-contrib-clean#rimraf#glob#minimatch"
   - in the nohoist list ["/_project_/**/@types/*","/_project_/**/@types/*/**","/_project_/**/grunt-*","/_project_/**/grunt-*/**","/_project_/@elastic/eui/rehype-react","/_project_/@elastic/eui/remark-rehype","/_project_/@elastic/eui/remark-rehype/**"]
info Disk size without dependencies: "44KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "156KB"
info Number of shared dependencies: 3
Done in 0.90s.
```

@kavilla do you mean this?

Reply:
> @manasvinibs if the problem is a direct dependency, of course updating the dep is necessary. When it is a nested dependency that needs bumping, sometimes you can get away with just changing the lockfile but sometimes you need to add overrides.
> 
> Here, bumping this dependency, doesn't do anything since in the lockfile, it is already resolved to `3.1.2`.

@AMoo-Miki Thanks Miki for explaining! So just to confirm here minimatch is a nested dependency through mocha, right?
```
 mocha@7.2.0
│ ├─┬ glob@7.1.3
│ │ └── minimatch@3.1.2 
│ └── minimatch@3.0.4 
```

Reply:
> Here, bumping this dependency, doesn't do anything since in the lockfile, it is already resolved to `3.1.2`.

@AMoo-Miki Does it mean, the fix should be removing minimatch@3.0.4 from yarn.lock through Mocha clean up? Also as it anyway resolves to 3.1.2 currently, is it still a valid CVE?

Reply:
Correct. Here, we have to add a resolution.